src/plugins/data_source_management/server/routes/data_connections_router.ts

```diff
@@ -291,4 +291,50 @@ export function registerDataConnectionsRoute(router: IRouter, dataSourceEnabled:
       }
     }
   );
+
+  router.delete(
+    {
+      path: `${DATACONNECTIONS_BASE}/{name}/dataSourceMDSId={dataSourceMDSId?}`,
+      validate: {
+        params: schema.object({
+          name: schema.string(),
+          dataSourceMDSId: schema.maybe(schema.string({ defaultValue: '' })),
```

Comment:
It's a query I suppose? And the path does not looks good to me.

Reply:
Just an example we did for data source id:
```javascript
{
  params: {...},
  query: schema.object({
    dataSourceId: schema.maybe(schema.string()),
  }),
}
```
then you can call the API with query string, for example `?dataSourceId=test`

Reply:
I'd say we can align with what we are having for the router for now (https://github.com/opensearch-project/OpenSearch-Dashboards/blob/f872b0255bc707fa45a076bdaff9d2930edc52df/src/plugins/data_source_management/server/routes/data_connections_router.ts#L251). And I agree that the path can be improved, and we can do that as a followup